GetStream/stream-chat-android
# [2238] Improve Quoted Messages

### 🎯 Goal

Update quoted message layouts to fit the design and add the scroll to quoted message on click.

### 🛠 Implementation details

Added new `QuotedAttachmentFactory`, `ImageAttachmentQuotedContent` and `FileAttachmentQuotedContent` which render the quoted attachments as well as updated the `QuotedMessage` layout.

If the message is in the list we will scroll to that item using the `focusedState` inside `MessageItemState`. If no item is found in the list we will request from the api to retrieve the message and scroll to it.

### 🎨 UI Changes

| Attachments before | Attachments after |
| --- | --- |
| ![Screenshot_20220506_164525](https://user-images.githubusercontent.com/38032787/167157374-aa72af11-5651-4685-b632-50d7a87753ea.png) | ![Screenshot_20220506_164419](https://user-images.githubusercontent.com/38032787/167157264-1fece77c-cdf5-4d05-8304-f51e3754a57b.png) |

<table>
<thead>
<tr>
<th>Scroll no loading</th>
<th>Scroll with loading</th>
<th>Scroll with loading, no loadMessageById change</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<video src="https://user-images.githubusercontent.com/38032787/168233270-165de6c2-b9d1-4406-b664-df229e318d61.mp4" controls="controls" muted="muted" />
</td>
<td>
<video src="https://user-images.githubusercontent.com/38032787/168233321-fee3057b-46ca-49df-9506-0a0124e160a1.mp4" controls="controls" muted="muted" />
</td>
<td>
<video src="https://user-images.githubusercontent.com/38032787/168233165-9c7080b6-fc74-443a-a3a9-b30f99d9ff9e.mp4" controls="controls" muted="muted" />
</td>
</tr>
</tbody>
</table>

### 🧪 Testing

1. Build the develop branch, quote messages and click them
2. Build this branch, quote messages and open them
3. Quoted messages with attachments should have small previews and clicking on them should scroll to their position

### ☑️Contributor Checklist

#### General

#### Code & documentation

### ☑️Reviewer Checklist

### 🎉 GIF

![giphy (1)](https://user-images.githubusercontent.com/38032787/167155532-8b7e3829-658b-45fd-b36e-485a9de4818d.gif)
